pyiceberg/table/__init__.py

```diff
@@ -434,6 +458,9 @@ def overwrite(
         if table_arrow_schema != df.schema:
             df = df.cast(table_arrow_schema)
 
+        with self.update_snapshot(snapshot_properties=snapshot_properties).delete() as delete_snapshot:
+            delete_snapshot.delete_by_predicate(overwrite_filter)
```

Comment:
My understanding is that, currently it will

- delete a datafile if all rows satisfy the `overwrite_filter`
- append the df to the table.

Is this the expected behavior? Looks like we could even change the below `overwrite` to `append` because we only `append_data_file`

I feel that it would be reasonable if `overwrite_filter` holds the same funcitonality as `delete_filter` below, that we will partially overwrite the data file if some of the rows matching the filter.

Reply:
If we need to delete a subset of the file, it will load the file, filter out the rows, and write it again. So we're removing an old file, and writing it again. Thus the overwrite

Reply:
Let me add a test-case to clarify this!